ui/app/rif/pages/rns/subdomains/index.js

```diff
@@ -4,36 +4,187 @@ import PropTypes from 'prop-types';
+      buttonNext: 'n-table-pagination-next',
+    },
+  },
+}
 
 class Subdomains extends Component {
 
   static propTypes = {
-    classes: PropTypes.any,
+    subdomain: PropTypes.object.isRequired,
+    domainName: PropTypes.string.isRequired,
+    pageName: PropTypes.string.isRequired,
+    redirectParams: PropTypes.any.isRequired,
+    selectedResolverAddress: PropTypes.string,
+    isOwner: PropTypes.bool,
+    newChainAddresses: PropTypes.array,
+    showPopup: PropTypes.func,
+    deleteSubdomain: PropTypes.func,
+    waitForListener: PropTypes.func,
+    showTransactionConfirmPage: PropTypes.func,
+    showThis: PropTypes.func,
+    showToast: PropTypes.func,
+    getSubdomains: PropTypes.func,
   }
 
   constructor (props) {
     super(props);
+    const resolvers = Object.assign([], GET_RESOLVERS());
     this.state = {
+      resolvers: resolvers,
     };
   }
 
+  openDeletePopup (subdomain) {
+    this.props.showPopup('Delete Subdomain', {
+      text: 'Are you sure you want to delete the subdomain ' + subdomain.name + '?',
+      confirmCallback: async () => {
+        const transactionListenerId = await this.props.deleteSubdomain(subdomain.domainName, subdomain.name);
+        this.props.waitForListener(transactionListenerId).then(transactionReceipt => {
+          this.props.getSubdomains(this.props.domainName)
+            .then(subdomains => {
+              this.props.showThis(
+                this.props.pageName,
+                {
+                  ...this.props.redirectParams,
+                  newSubdomains: subdomains,
+                });
+              });
+            });
+        this.props.showTransactionConfirmPage({
+          action: () => {
+            this.props.showThis(
+              this.props.pageName,
+              this.props.redirectParams);
+            this.props.showToast('Waiting for confirmation');
+          },
+        });
+      },
+      confirmButtonClass: '',
+    });
+  }
+
   render () {
+    const { subdomain, domainName, selectedResolverAddress, newChainAddresses, isOwner } = this.props;
+    const displayName = domainName + '.' + subdomain.name;
+    const { resolvers } = this.state;
     return (
-      <div>Subdomains</div>
+      <div>
+        <DomainHeader
+          domainName={displayName}
+          showOwnerIcon={isOwner}
+        >
+          {isOwner &&
+            <CustomButton
+              text={'Delete'}
+              onClick={() => this.openDeletePopup(subdomain)}
+              className={
+                {
+                  button: '',
+                  icon: '',
+                  text: '',
+                }
+              }
+            />
+          }
+        </DomainHeader>
+        {resolvers &&
+        <div id="chainAddressesBody">
+          <ChainAddresses
+            domainName={domainName}
+            subdomainName={subdomain.name}
+            selectedResolverAddress={selectedResolverAddress}
+            paginationSize={3}
```

Comment:
Why is this 3? can you explain? it's an arbitrary value seleted or some specification, i recommend you to put that value in a constant.